compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1182,17 +1177,31 @@ class TreeUnpickler(reader: TastyReader,
               val levels = readNat()
               readTerm().outerSelect(levels, SkolemType(readType()))
             case SELECTin =>
-              var sname = readName()
+              var symname = readName()
               val qual = readTerm()
-              val owner = readType()
-              def select(name: Name, denot: Denotation) =
-                val prefix = ctx.typeAssigner.maybeSkolemizePrefix(qual.tpe.widenIfUnstable, name)
-                makeSelect(qual, name, denot.asSeenFrom(prefix))
-              sname match
+              val qualType = qual.tpe.widenIfUnstable
+              val space = readType()
+              def select(name: Name, sig: Signature, target: Name) =
+                def accessibleDenot(qualType: Type, name: Name, sig: Signature, target: Name) = {
+                  val pre = ctx.typeAssigner.maybeSkolemizePrefix(qualType, name)
+                  val d1 = qualType.findMember(name, pre)
+                  if !d1.isOverloaded && d1.infoOrCompleter.isInstanceOf[MethodOrPoly] then
+                    // workaround for method refinements that either override param names,
```

Comment:
Instead of adding such a workaround, I think it'd be fine to keep SELECT around for the rare cases where we need to do non-symbol selections.

Reply:
I still need to address this part